.github/workflows/frontend_ci_cd.yml

```diff
@@ -133,3 +133,36 @@ jobs:
         with:
           name: cypress-screenshots
           path: ${{github.workspace}}/frontend/cypress/screenshots
+
+  deploy:
+    name: Deploy frontend
+    if: (github.head_ref || github.ref_name) == 'develop'
+    runs-on: ubuntu-22.04
+    timeout-minutes: 30
+    needs: [e2e-chrome, e2e-firefox]
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Install ruby + gems
+        uses: ruby/setup-ruby@v1
+        with:
+          # ruby-version: 3.2.1 # Not needed with a .ruby-version file
+          bundler-cache: true
+          working-directory: frontend
+
+      - name: Extract branch name
+        shell: bash
+        run: |
+          echo "##[set-output name=branch;]$(echo ${GITHUB_REF#refs/heads/})"
+          echo "##[set-output name=branch-upper;]$(echo ${GITHUB_REF#refs/heads/} | tr a-z A-Z )"
+        id: extract_branch
+
+      - name: Deploy using capistrano
+        uses: miloserdow/capistrano-deploy@v3
+        with:
+          working-directory: frontend
+          target: ${{ steps.extract_branch.outputs.branch == 'master' && 'production' || 'staging' }}
```

Comment:
Here you're checking if the branch is `master` but on line 139 you said the job only runs when the branch is `develop`.

Reply:
This is related to the target of Capistrano, which they are only two options: staging and production

Reply:
yes, this should stay like this, because once we enable the workflow on master as well as develop this will be needed